http/digest/src/main/java/org/wildfly/security/http/digest/NonceManager.java

```diff
@@ -59,6 +52,22 @@ public class NonceManager {
     private final String algorithm;
     private ElytronMessages log;
 
+    /**
+     * initialize with default values
+     */
+    NonceManager() {
```

Comment:
@fjuma Since the PersistentNonceManager is Serializable, it requires that its parent NonceManager to have a no-arg constructor, otherwise it won't compile. Alternative would be to make NonceManager Serializable and make its ScheduledExecutorService variable `transient`.